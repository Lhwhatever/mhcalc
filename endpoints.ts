export interface Endpoint {
    name: string
    path: string
}

const endpoints = {
    calculators: [
        {
            name: 'Valour Rift Simulator',
            path: '/vrift'
        }
    ]
}

export default endpoints
