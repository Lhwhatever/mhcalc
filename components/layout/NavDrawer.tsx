import {
    Box,
    Divider,
    IconButton,
    List,
    makeStyles,
    SwipeableDrawer,
    SwipeableDrawerProps,
    Theme
} from '@material-ui/core'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import React from 'react'
import { Endpoint } from '../../endpoints'
import ListLinkItem, { ListLinkItemProps } from '../link/ListLinkItem'
import DrawerSubList from './DrawerSubList'

const useStyles = makeStyles((theme: Theme) => ({
    drawerHeader: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-end',
        padding: theme.spacing(0, 1),
        ...theme.mixins.toolbar,
        justifyContent: 'center'
    }
}))

export interface NavDrawerEndpoints {
    calculators: Endpoint[]
}

interface NavDrawerExtendedProps {
    endpoints: NavDrawerEndpoints
    current?: string
}

export type NavDrawerProps = SwipeableDrawerProps & NavDrawerExtendedProps

const NavDrawer = (props: NavDrawerProps): JSX.Element => {
    const { endpoints, current, ...other } = props

    const classes = useStyles()

    const UngroupedEndpoint = (props: ListLinkItemProps) => <ListLinkItem {...props} active={current === props.href} />

    return (
        <SwipeableDrawer {...other}>
            <Box className={classes.drawerHeader}>
                <IconButton onClick={props.onClose} aria-label="close">
                    <ChevronLeftIcon />
                </IconButton>
            </Box>
            <Box component="div" overflow="auto">
                <List component="nav" aria-label="navigation drawer">
                    <UngroupedEndpoint href="/">Home</UngroupedEndpoint>
                    <Divider />
                    <DrawerSubList subheader="Calculators" endpoints={endpoints.calculators} current={current} />
                </List>
            </Box>
        </SwipeableDrawer>
    )
}

export default NavDrawer
