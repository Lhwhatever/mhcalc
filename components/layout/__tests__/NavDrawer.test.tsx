import { Box, IconButton, SwipeableDrawer } from '@material-ui/core'
import { createShallow } from '@material-ui/core/test-utils'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import React from 'react'
import endpoints from '../../../endpoints'
import NavDrawer from '../NavDrawer'

const noOp = () => undefined

describe('NavDrawer test', () => {
    let shallow
    let wrapper

    beforeAll(() => {
        shallow = createShallow()
    })

    beforeEach(() => {
        wrapper = shallow(<NavDrawer endpoints={{}} open={true} onOpen={noOp} onClose={noOp} />)
    })

    it('should use SwipeableDrawer', () => {
        expect(wrapper.find(SwipeableDrawer)).toHaveLength(1)
    })

    it('should have a close button', () => {
        expect(
            wrapper
                .find(Box)
                .at(0)
                .containsMatchingElement(
                    <IconButton>
                        <ChevronLeftIcon />
                    </IconButton>
                )
        ).toBe(true)
    })

    it('should render correctly', () => {
        wrapper.setProps({ endpoints })
        expect(wrapper).toMatchSnapshot()
    })
})
