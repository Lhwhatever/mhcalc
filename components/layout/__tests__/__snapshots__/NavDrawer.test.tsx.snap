// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer test should render correctly 1`] = `
<ForwardRef(SwipeableDrawer)
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  open={true}
>
  <Styled(MuiBox)
    className="makeStyles-drawerHeader-1"
  >
    <WithStyles(ForwardRef(IconButton))
      aria-label="close"
      onClick={[MockFunction]}
    >
      <ChevronLeftIcon />
    </WithStyles(ForwardRef(IconButton))>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    component="div"
    overflow="auto"
  >
    <WithStyles(ForwardRef(List))
      aria-label="navigation drawer"
      component="nav"
    >
      <UngroupedEndpoint
        href="/"
      >
        Home
      </UngroupedEndpoint>
      <WithStyles(ForwardRef(Divider)) />
      <DrawerSubList
        component="div"
        endpoints={
          Array [
            Object {
              "name": "foo",
              "path": "/foo",
            },
            Object {
              "name": "bar",
              "path": "/bar",
            },
          ]
        }
        subheader="Calculators"
      />
    </WithStyles(ForwardRef(List))>
  </Styled(MuiBox)>
</ForwardRef(SwipeableDrawer)>
`;
